metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/utils/EntityDao.java

```diff
@@ -0,0 +1,263 @@
+    List<MongoTerm> mongoTerms = createListOfMongoTerms(entity);
+    saveMongoTerms(mongoTerms, entityType);
+    return mongoTerms.size();
+  }
+
+
+  private static List<MongoTerm> createListOfMongoTerms(ContextualClassImpl entity) {
+    MongoTerm mongoTerm;
+    List<MongoTerm> mongoTerms = new ArrayList<>();
+
+    for (Map.Entry<String, List<String>> prefLabel : entity.getPrefLabel().entrySet()) {
+      for (String label : prefLabel.getValue()) {
+        mongoTerm = new MongoTerm();
+        mongoTerm.setCodeUri(entity.getAbout());
+        mongoTerm.setLabel(label.toLowerCase());
+        mongoTerm.setOriginalLabel(label);
+        mongoTerm.setLang(prefLabel.getKey());
+        mongoTerms.add(mongoTerm);
+      }
+    }
+    return mongoTerms;
+  }
+
+  public List<String> deleteAllEntitiesMatching(List<String> codeUris) {
+    List<String> removedCodeUris = new ArrayList<>();
+    for (String codeUri : codeUris) {
+      removedCodeUris.add(codeUri);
+      removedCodeUris.addAll(deleteEntities(getTableName(EntityType.PLACE), PLACE_TYPE, codeUri));
+      removedCodeUris
+          .addAll(deleteEntities(getTableName(EntityType.CONCEPT), CONCEPT_TYPE, codeUri));
+      removedCodeUris.addAll(deleteEntities(getTableName(EntityType.AGENT), AGENT_TYPE, codeUri));
+      removedCodeUris
+          .addAll(deleteEntities(getTableName(EntityType.TIMESPAN), TIMESPAN_TYPE, codeUri));
+      removedCodeUris.addAll(
+          deleteEntities(getTableName(EntityType.ORGANIZATION), ORGANIZATION_TYPE, codeUri));
+    }
+    return removedCodeUris;
+  }
+
+  public List<String> deleteEntities(String entityTable, String entityType, String codeUri) {
+    List<String> extraUrisRemoved = new ArrayList<>();
+    //Remove from Term List
+    deleteMongoTermList(codeUri);
+    //Remove from specific collection
+    deleteMongoTerm(entityTable, codeUri);
+
+    //Find all TermLists that have owlSameAs equals with codeUri
+    final Query<MongoTermList> termListsSameAsQuery = this.datastore
+        .createQuery(MongoTermList.class).filter(ENTITY_TYPE_FIELD, entityType)
+        .filter(OWL_SAME_AS, codeUri);
+    final List<MongoTermList> allTermListsSameAs = getListOfQuery(termListsSameAsQuery, null);
+    for (MongoTermList mongoTermList : allTermListsSameAs) {
+      final String sameAsCodeUri = mongoTermList.getCodeUri();
+      extraUrisRemoved.add(sameAsCodeUri);
+      //Remove from Term List
+      deleteMongoTermList(codeUri);
+      //Remove from specific collection
+      deleteMongoTerm(entityTable, sameAsCodeUri);
+    }
+    return extraUrisRemoved;
+  }
+
+  public void deleteMongoTerm(String entityTable, String codeUri) {
+    final MongoTerm mongoTermCoreUri = findMongoTermByField(entityTable, CODE_URI_FIELD, codeUri);
+    ExternalRequestUtil.retryableExternalRequestConnectionReset(
+        () -> this.datastore.delete(entityTable, MongoTerm.class, mongoTermCoreUri.getId()));
```

Comment:
delete is deprecated.

Reply:
Yes there is no other way at the moment, we'll need to change the db structure to fix this